src/coreclr/vm/jitinterface.cpp

```diff
@@ -2478,6 +2512,23 @@ void CEEInfo::MethodCompileComplete(CORINFO_METHOD_HANDLE methHnd)
         pMD->AsDynamicMethodDesc()->GetResolver()->FreeCompileTimeState();
     }
 
+    // Free all handles used by JIT
```

Comment:
This would be leaking the handles when the compilation fails. It should be in destructor instead. `MethodCompileComplete` is only called on when the compilation succeeds.